From ff468584abddcb3f703d407aecf78137fb656472 Mon Sep 17 00:00:00 2001
From: Evgeniy Khramtsov <evgeniy@khramtsov.org>
Date: Fri, 1 Oct 2021 16:11:39 +0300
Subject: [PATCH] commands: Remove unused code after 1d3681f52135

Clang 13 reports:

../sway/commands.c:470:23: error: variable 'context' set but not used
[-Werror,-Wunused-but-set-variable]
        enum command_context context = 0;
                             ^
Last use of was removed in commit 1d3681f52135.

Downstream PR: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=258813
---
 include/sway/config.h |  8 --------
 sway/commands.c       | 28 ----------------------------
 2 files changed, 36 deletions(-)

diff --git a/include/sway/config.h b/include/sway/config.h
index 52867fa665..46dd4ffe4d 100644
--- a/include/sway/config.h
+++ b/include/sway/config.h
@@ -418,14 +418,6 @@ enum sway_popup_during_fullscreen {
 	POPUP_LEAVE,
 };
 
-enum command_context {
-	CONTEXT_CONFIG = 1 << 0,
-	CONTEXT_BINDING = 1 << 1,
-	CONTEXT_IPC = 1 << 2,
-	CONTEXT_CRITERIA = 1 << 3,
-	CONTEXT_ALL = 0xFFFFFFFF,
-};
-
 enum focus_follows_mouse_mode {
 	FOLLOWS_NO,
 	FOLLOWS_YES,
diff --git a/sway/commands.c b/sway/commands.c
index b09a04c71d..205406ad2a 100644
--- a/sway/commands.c
+++ b/sway/commands.c
@@ -465,34 +465,6 @@ struct cmd_results *config_commands_command(char *exec) {
 		goto cleanup;
 	}
 
-	enum command_context context = 0;
-
-	struct {
-		char *name;
-		enum command_context context;
-	} context_names[] = {
-		{ "config", CONTEXT_CONFIG },
-		{ "binding", CONTEXT_BINDING },
-		{ "ipc", CONTEXT_IPC },
-		{ "criteria", CONTEXT_CRITERIA },
-		{ "all", CONTEXT_ALL },
-	};
-
-	for (int i = 1; i < argc; ++i) {
-		size_t j;
-		for (j = 0; j < sizeof(context_names) / sizeof(context_names[0]); ++j) {
-			if (strcmp(context_names[j].name, argv[i]) == 0) {
-				break;
-			}
-		}
-		if (j == sizeof(context_names) / sizeof(context_names[0])) {
-			results = cmd_results_new(CMD_INVALID,
-					"Invalid command context %s", argv[i]);
-			goto cleanup;
-		}
-		context |= context_names[j].context;
-	}
-
 	results = cmd_results_new(CMD_SUCCESS, NULL);
 
 cleanup:
